forum/projects/services.py

```diff
@@ -28,3 +28,20 @@ def notify_investors_via_email(project, changes):
         sender=settings.EMAIL_HOST_USER,
         receivers=recipients,
     )
+
+def send_notification(self, project, action):
+        subject = f"Project {action.capitalize()}"
+        message = build_email_message(
+            "email/project_update_notification.txt",  
+            {
+                "project_name": project.name,
+                "action": action,

```

Comment:
Validate inputs to ensure `project` and `action` have the expected attributes. This prevents potential issues if the data structure changes.

Reply:
done!